core/src/test/java/org/apache/iceberg/rest/TestRESTCatalog.java

```diff
@@ -2329,6 +2374,57 @@ public void multipleDiffsAgainstMultipleTablesLastFails() {
     assertThat(schema2.columns()).hasSize(1);
   }
 
+  @Test
+  public void testInvalidRestPageSize() {
+    RESTCatalogAdapter adapter = Mockito.spy(new RESTCatalogAdapter(backendCatalog));
+    RESTCatalog catalog =
+            new RESTCatalog(SessionCatalog.SessionContext.createEmpty(), (config) -> adapter);
+    org.assertj.core.api.Assertions.assertThatThrownBy(
+            () -> catalog.initialize("test", ImmutableMap.of(RESTSessionCatalog.REST_PAGE_SIZE, "-1")))
+            .isInstanceOf(IllegalArgumentException.class)
+            .hasMessage("Invalid value for pageSize, must be a positive integer");
+  }
+
+  @Test
+  public void testPaginationForListNamespaces() {
+    RESTCatalogAdapter adapter = Mockito.spy(new RESTCatalogAdapter(backendCatalog));
```

Comment:
no need for `Mockito.spy()` here as you're not doing any verifications on the `adapter` in this test (and in the other tests)